Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp

```diff
@@ -791,15 +916,13 @@ namespace AzFramework
 #ifdef CARBONATED
                     // We should add that into m_entitySpawnableMap before "Events::AddGameEntity" because Activate is called there and we
                     // need this data there
-                    if (!ticket.m_spawnable->GetInstanceId().IsNull())
+                    if (spawnableInfo && spawnableInfo->IsValid())
                     {
-                        m_entitySpawnableMap.insert(AZStd::make_pair(
-                            clone->GetId(), SpawnableInstanceAddress(ticket.m_spawnable.GetId(), ticket.m_spawnable->GetInstanceId())));
+                        m_entitySpawnableMap.insert(AZStd::make_pair(clone->GetId(), spawnableInfo));
```

Comment:
Do we need  "                    clone->SetEntitySpawnTicketId(request.m_ticketId);" here?

Reply:
Yes, we need to call "clone->SetEntitySpawnTicketId(request.m_ticketId);" at least once here (it is an original code).
Refactored this place to avoid misleads.

Also removed duplicated "clone->SetEntitySpawnTicketId(request.m_ticketId);" call in another place, after line 740